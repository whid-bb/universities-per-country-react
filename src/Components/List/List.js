import { useSelector } from 'react-redux';

import ListItem from './ListItem/ListItem';
import './List.scss';

function List() {
  const { data } = useSelector((state) => state.countries);
  return (
    <div>
      <ListItem country={data} />
    </div>
  );
}

export default List;
