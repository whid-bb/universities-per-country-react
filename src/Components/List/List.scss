$darkerBlue: #4369b2;

.country-list-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: stretch;
  // align-items: center;
  justify-content: center;

  .row {
    flex: 1 100%;
    display: flex;
    align-items: stretch;
    flex-direction: row;
    flex-wrap: wrap;
    &:nth-child(2n) {
      flex-direction: row-reverse;
    }
  }
  .country-list-item {
  }
}
