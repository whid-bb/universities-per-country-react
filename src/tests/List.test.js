import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import List from '../Components/List/List';

describe('List test', () => {
  it('List render correctly', () => {
    const ListPage = renderer.create(
      <Provider store={store}>
        <Router>

          <List />
        </Router>
      </Provider>,
    ).toJSON();
    expect(ListPage).toMatchSnapshot();
  });

  test('Render List component', () => {
    render(
      <Provider store={store}>
        <Router>

          <List />
        </Router>
      </Provider>,
    );
  });
});
