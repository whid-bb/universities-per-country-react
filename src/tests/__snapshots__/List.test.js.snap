// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test List render correctly 1`] = `
<div>
  <div
    className="country-list-container"
  />
</div>
`;
